api/registry.py

```diff
@@ -384,3 +406,79 @@ def _process_registration_result(self, catalog, encryptor, desired_stage):
         # communicated to the registry.
         self.stage_field.value = desired_stage
         return True
+
+
+class LibraryRegistrationScript(LibraryInputScript):
+    """Register local libraries with a remote library registry."""
+
+    PROTOCOL = ExternalIntegration.OPDS_REGISTRATION
+    GOAL = ExternalIntegration.DISCOVERY_GOAL
+
+    @classmethod
+    def arg_parser(cls, _db):
+        parser = LibraryInputScript.arg_parser(_db)
+        parser.add_argument(
+            '--registry-url',
+            help="Register libraries with the given registry.",
+            default=RemoteRegistry.DEFAULT_LIBRARY_REGISTRY_URL
+        )
+        parser.add_argument(
+            '--testing',
+            help="Tell the registry that you do not consider these libraries ready for production.",
+            action='store_true'
+        )
```

Comment:
The default is "production" because in general it'll be more dangerous to set a production library to "testing" than to set a library to "production" before you're ready. A library doesn't go into SimplyE production until both the library and the registry think it should. Whereas taking a library from "production" to "testing" can immediately remove it from view for most SimplyE users.

However I'm not sure we want it to be possible to put a library in "testing" once it goes into production, so this could change in the future.

Reply:
Yeah, I would make it so the default is "testing" if the library isn't registered yet, but if it's in production it stays there.

Reply:
Thanks, this is a really good idea, I just had to sit with this a while to figure out how to do it.